internal/controllers/image/actuator.go

```diff
@@ -85,12 +85,33 @@ func (actuator imageActuator) ListOSResourcesForAdoption(ctx context.Context, ob
 		return nil, false
 	}
 
-	listOpts := images.ListOpts{Name: string(getResourceName(obj))}
-	return actuator.osClient.ListImages(ctx, listOpts), true
+	listOpts := images.ListOpts{
+		Name: string(getResourceName(obj)),
+	}
+
+	if len(obj.Spec.Resource.Tags) > 0 {
+		listOpts.Tags = make([]string, len(obj.Spec.Resource.Tags))
+		for i := range obj.Spec.Resource.Tags {
+			listOpts.Tags[i] = string(obj.Spec.Resource.Tags[i])
+		}
+	}
+
+	existingImage := actuator.osClient.ListImages(ctx, listOpts)
+	return existingImage, true
 }
 
 func (actuator imageActuator) ListOSResourcesForImport(ctx context.Context, filter filterT) imageIterator {
-	listOpts := images.ListOpts{Name: string(ptr.Deref(filter.Name, ""))}
+	listOpts := images.ListOpts{
+		Name: string(ptr.Deref(filter.Name, "")),
```

Comment:
Aside: did we check if this does a substring match like nova?

Reply:
It does not. I've added a negative check, for both adoption and import. I'll later port that change to the other `import` tests we have already written.